src/js/components/bulkDownload/awards/filters/AwardLevelFilter.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * AwardLevelFilter.jsx
+ * Created by Lizzie Salita 11/2/17
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { CheckCircle, ExclamationTriangle } from 'components/sharedComponents/icons/Icons';
+import { indexOf, remove, concat } from 'lodash';
+
+import DownloadCheckbox from '../../DownloadCheckbox';
+
+const propTypes = {
+    awardLevels: PropTypes.array,
+    currentAwardLevels: PropTypes.array,
+    updateParam: PropTypes.func
+};
+
+export default class AwardLevelFilter extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.onChange = this.onChange.bind(this);
+    }
+
+    onChange(name, value) {
+        if (value) {
+            const updatedAwardLevels = concat(this.props.currentAwardLevels, [name]);
+            this.props.updateParam('award_levels', updatedAwardLevels);
+        }
+        else {
+            const updatedAwardLevels = remove(this.props.currentAwardLevels, (item) => item !== name);
+            this.props.updateParam('award_levels', updatedAwardLevels);
+        }
+    }
+
+    render() {
+        const awardLevelsArray = this.props.currentAwardLevels;
+        const currentAwardLevels = {
```

Comment:
Why are we handling these types as arrays if we are performing this many search operations with the array? Why not use an object or other keyed data structure?

Reply:
Same comment applies to the above `onChange` function